tmva/sofie/inc/TMVA/ROperator_Range.hxx

```diff
@@ -0,0 +1,74 @@
+#ifndef TMVA_SOFIE_ROPERATOR_Range
+#define TMVA_SOFIE_ROPERATOR_Range
+
+#include "TMVA/SOFIE_common.hxx"
+#include "TMVA/ROperator.hxx"
+#include "TMVA/RModel.hxx"
+
+#include <sstream>
+#include <cmath>
+
+namespace TMVA {
+namespace Experimental {
+namespace SOFIE {
+
+template <typename T>
+class ROperator_Range final : public ROperator {
+
+private:
+   std::string fNStart;
+   std::string fNLimit;
+   std::string fNDelta;
+   std::size_t num_of_elem;
+   std::string fNY;
+   std::vector<size_t> fShapeY;
+
+public:
+   ROperator_Range() {}
+
+   ROperator_Range(std::string nameStart, std::string nameLimit, std::string nameDelta, std::string nameY)
+      : fNStart(nameStart), fNLimit(nameLimit), fNDelta(nameDelta), fNY(UTILITY::Clean_name(nameY))
+   {
+   }
+
+   std::vector<ETensorType> TypeInference(std::vector<ETensorType> input) { return input; }
+
+   std::vector<std::vector<size_t>> ShapeInference(std::vector<std::vector<size_t>> input)
+   {
+      auto ret = input; // suggest copy to compiler
+      return ret;
+   }
+   void Initialize(RModel &model)
+   {
+      // input must be a scalar
+      double lim = std::stod(fNLimit);
```

Comment:
Are you sure the inputs to the Range operator in ONNX are actually a `double` value but in a string? I thought it would have a string name to represent in the model, and it will take up values as passed or present in the model.